libs/platform/ebpf_object.h

```diff
@@ -9,6 +9,87 @@ extern "C"
 {
 #endif
 
+    /**
+     * @brief Identifier for the file that is referencing the object. Each file that references an object is assigned
+     * a unique identifier. This identifier is used to track the reference count of the object. The file identifier is
+     * recorded in the _ebpf_object_reference_history table along with the line number of the reference. This allows
+     * for tracking down use-after-free bugs and leaks of objects.
+     */
+    typedef enum _ebpf_file_id
+    {
+        EBPF_FILE_ID_UNKNOWN,
+        EBPF_FILE_ID_CORE,
+        EBPF_FILE_ID_MAPS,
+        EBPF_FILE_ID_LINK,
+        EBPF_FILE_ID_PROGRAM,
+        EBPF_FILE_ID_NATIVE,
+        EBPF_FILE_ID_PINNING_TABLE,
+        EBPF_FILE_ID_HANDLE,
+        EBPF_FILE_ID_EXECUTION_CONTEXT_UNIT_TESTS,
+        EBPF_FILE_ID_PLATFORM_UNIT_TESTS,
+        EBPF_FILE_ID_PERFORMANCE_TESTS,
+        EBPF_FILE_ID_CORE_HELPER_FUZZER,
+    } ebpf_file_id_t;
+
+/**
+ * @brief Macro to acquire a reference on an object and record the file and line number of the reference.
+ */
+#define EBPF_OBJECT_ACQUIRE_REFERENCE(object) ebpf_object_acquire_reference(object, EBPF_FILE_ID, __LINE__)
```

Comment:
question: would the __LINE__ would if this were an inline function instead of a macro?